android/src/main/java/com/baseflow/permissionhandler/PermissionHandlerPlugin.java

```diff
@@ -249,10 +268,23 @@ private int checkPermissionStatus(@PermissionGroup int permission) {
       return PERMISSION_STATUS_UNKNOWN;
     }
 
-    final boolean targetsMOrHigher = context.getApplicationInfo().targetSdkVersion >= android.os.Build.VERSION_CODES.M;
+    final boolean targetsMOrHigher = context.getApplicationInfo().targetSdkVersion >= VERSION_CODES.M;
 
     for (String name : names) {
       if (targetsMOrHigher) {
+        if (permission == PERMISSION_GROUP_IGNORE_BATTERY_OPTIMIZATIONS) {
+          String packageName = context.getPackageName();
+          PowerManager pm = (PowerManager) context.getSystemService(Context.POWER_SERVICE);
+          if (VERSION.SDK_INT >= VERSION_CODES.M) {
```

Comment:
This `if` statement seems redundant since there is already a check which checks if the current Android version is M or higher on line 274.

Reply:
The first check only covers the Apps AndroidManifest.xml. The second check (Version.SDK_INT) is a runtime check for the actual Android version currently running. Therefore, I'd suggest to comment & stick with the way.

WIP: not anymore for our side, as it's in production apps.